### PR TITLE
Reland "[NFC][AMDGPU] Do not flush after printing every instruction"

### PR DESCRIPTION
**Pull request by Pierre-vh**
Reland of #95237 with fix to failing test. The test relied on stderr/stdout output being interleaved in a specific order.

It's very expensive and doesn't achieve anything.

I one test I did, it saves almost 10s on a 2m23s build, bringing it down to 2m15s using a downstream branch.

### CODE REVIEW

File: llvm/test/MC/AMDGPU/hsa-sym-expr-failure.s [hunk 1]
```diff
@@ -49,12 +44,10 @@ user_sgpr_kernarg_preload_length:
   .amdhsa_next_free_vgpr 0
   .amdhsa_next_free_sgpr 0
   .amdhsa_accum_offset 4
+// ASM: :[[@LINE+1]]:{{[0-9]+}}: error: directive should have resolvable expression
```
Comment (arsenm) [inline llvm/test/MC/AMDGPU/hsa-sym-expr-failure.s:47]: Just precommit the unrelated test changes

**Comment by Pierre-vh** [comment]
fc19d6238af7d215b13c2af63cbce03b5c04069b and fc19d6238af7d215b13c2af63cbce03b5c04069b
